### PR TITLE
update-build-tools-branch

### PR DESCRIPTION
**Pull request by luxnova**
## What Changed
Updated the build tools and modified some of the grade files to bring the library up to date. 

## Review
- [x] @brentwatson

### CODE REVIEW

File: caffeine/build.gradle [hunk 1]
```diff
@@ -5,7 +5,6 @@ android {
     buildToolsVersion "20.0.0"
```
Comment (brentwatson) [inline caffeine/build.gradle]: Let's bump up `compileSdkVersion`, `buildToolsVersion`, and `targetSdkVersion` too.

File: caffeine/build.gradle [hunk 2]
```diff
@@ -38,4 +37,3 @@ task makeJar(type: Copy) {
 
 makeJar.dependsOn(build)
 
-apply from: 'maven_push.gradle'
```
Comment (brentwatson) [inline caffeine/build.gradle]: This line is still needed.

File: .idea/.name [hunk 3]
```diff
@@ -1 +1 @@
-Caffeine Project
```
Comment (lra) [inline .idea/.name]: Do we need `.idea` in git?

Comment (brentwatson) [inline .idea/.name]: That's a toss-up.  Doesn't hurt to have the .idea dir.  It means you can "Open Project" in the IDE vs import it, which is easier for external people to contribute to this library.

The downside is that Android Studio makes useless changes to those files, which end up in PRs.  We usually exclude .idea changes, but this PR was to update them.  

FWIW, our internal repos ignore `.idea`.

Comment (lra) [inline .idea/.name]: yep, not sure if it's a sign of good engineering practice to do this on a public repo.
also, I'm always worried about sensible info and hardcoded path in those config files.

Your call.

Comment (brentwatson) [inline .idea/.name]: Checked some other public repos, and most do in fact ignore `.idea` dir and make import vs open the project.  I'll PR to remove it.

Comment (brentwatson) [inline .idea/.name]: :boom: https://github.com/percolate/caffeine/pull/12

**Pull request merged**